### PR TITLE
[extractor/rtl.nl] Add rtl.lu extractor

### LINKED ISSUE #1721: rtl.lu site request
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2021.11.10.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

europe

### Example URLs

https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html

### Description

Thanks for adding this additional site if possible.

### Verbose log

```shell
D:\Programmes\Tools\Batch\FFMPEG>yt-dlp.exe -Uv https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html
[debug] Command-line config: ['-Uv', 'https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.11.10.1 (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.4-essentials_build-www.gyan.dev (setts), ffprobe 4.4-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
yt-dlp is up to date (2021.11.10.1)
[debug] [generic] Extracting URL: https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html
[generic] 3266757: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] 3266757: Downloading webpage
[generic] 3266757: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1311, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1336, in __extract_info
  File "yt_dlp\extractor\common.py", line 595, in extract
  File "yt_dlp\extractor\generic.py", line 3766, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.rtl.lu/tele/de-journal-vun-der-tele/v/3266757.html
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

Add support for www.rtl.lu (and today.lu, 5minutes.lu for `Article` type). This PR adds extractor for Tele-VOD(Tele-Replay), Tele-Live, video or audio in `article` type, and audio in `Radio`
Close #1721

### CODE REVIEW

File: yt_dlp/extractor/rtllu.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    _MEDIA_REGEX = {
+        'video': r'<rtl-player\s*poster\s*=\s*\"(?P<thumbnail_url>[\"\w\.://-]+)\"\s*title\s*=\s*(?P<title>[\"\w\.-]+)\s*hls\s*=\s*\"(?P<media_url>[\w\.\:/-]+)\"',
+        'audio': r'<rtl-audioplayer\s*src\s*=\s*\"(?P<media_url>[\w\.\:/-]+)\"',
+    }
+
+    def get_media(self, media_type, webpage, video_id):
+        media_url = self._search_regex(
+            self._MEDIA_REGEX[media_type], webpage, 'media_url', group=('media_url'))
+
+        return media_url
```
Comment (pukkandan) [inline yt_dlp/extractor/rtllu.py:14]: 1. What is the point of this BaseIE? 
2. You are only using `media_type='video'`. What's audio for?
3. Why are you capturing, but discarding thumbnail_url?

This just looks incomplete...

Comment (HobbyistDev) [reply yt_dlp/extractor/rtllu.py:14]: Actually, i want to extend this extractor to extract audio in `rtl.lu/radio` and audio/video in `rtl.lu/.../news` but i think it's better if all of these be split to another PR.

**Comment by HobbyistDev** [comment]
No, it's different

File: yt_dlp/extractor/rtllu.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/rtllu.py:14]: The problem is, I can't judge whether the code should be structured like this by seeing just this. You could trim down this PR to only what is needed now, but that would be unnecessary work imo. It is better to complete the extractor

**Comment by HobbyistDev** [comment]
After searching in wikipedia, i think they are related. I will move this to `rtl.nl`

File: yt_dlp/extractor/rtlnl.py [hunk 2]
```diff
@@ -141,3 +141,176 @@ def _real_extract(self, url):
...
+        'video': r'<rtl-player\s*poster\s*=\s*\"(?:[\"\w\.://-]+)\"\s*(title\s*=\s*(?:[\"\w\.-]+))?\s*(type\s*=\s*\"(?:\w+)\")?\s*(channelname\s*=\s*\"(?:\w+)\")?\s*hls\s*=\s*\"(?P<media_url>[\w\.\:/-]+)\"',
+        'audio': r'<rtl-audioplayer\s*src\s*=\s*\"(?P<media_url>[\w\.\:/-]+)\"',
+    }
+
+    def get_media_url(self, webpage, video_id, media_type):
+        media_url = self._search_regex(
+            self._MEDIA_REGEX[media_type], webpage, 'media_url', group=('media_url'),
+            default=None, fatal=False)
+
+        return media_url
```
Comment (Ashish0804) [inline yt_dlp/extractor/rtlnl.py:157]: inline

File: yt_dlp/extractor/rtlnl.py [hunk 3]
```diff
@@ -141,3 +141,176 @@ def _real_extract(self, url):
...
+
+    def get_format(self, webpage, video_id):
+        video_url, audio_url = self.get_media_url(webpage, video_id, 'video'), self.get_media_url(webpage, video_id, 'audio')
+
+        formats, subtitles = [], {}
+        if video_url is not None:
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, video_id)
+        if audio_url is not None:
+            audio_format = {'url': audio_url, 'ext': 'mp3'}
+            formats.append(audio_format)
```
Comment (Ashish0804) [inline yt_dlp/extractor/rtlnl.py:167]: inline

File: yt_dlp/extractor/rtlnl.py [hunk 4]
```diff
@@ -141,3 +141,176 @@ def _real_extract(self, url):
...
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnail': self._og_search_thumbnail(webpage),
+        }
+
+
+class RTLLuTeleLiveIE(RTLLuBaseIE):
+    _VALID_URL = r'https?://www\.rtl\.lu/tele/live(?P<id>-\d+)?'
```
Comment (Ashish0804) [inline yt_dlp/extractor/rtlnl.py:256]: this is incorrect imo. 
U are calling the group id here and it is `?`.

Either u should rename it to slug (which again i do not prefer)
Or you should make the whole group capture compulsory and then u can call it id.(which i prefer) 

`r'https?://www\.rtl\.lu/tele/(?P<id>live(?:-\d+)?)'`

File: yt_dlp/extractor/rtlnl.py [hunk 5]
```diff
@@ -141,3 +141,176 @@ def _real_extract(self, url):
...
+            'id': 'Tele:live-2',
+            'ext': 'mp4',
+            'live_status': 'is_live',
+            'title': r're:RTL - Télé LIVE \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+        }
+    }]
+
+    def _real_extract(self, url):
+        slug = '' if self._match_id(url) is None else self._match_id(url)
+        video_id = f'Tele:live{slug}'
```
Comment (Ashish0804) [inline yt_dlp/extractor/rtlnl.py:278]: not a good idea to set id manually imo.
Just apply above suggestion and do video_id = self._match_id(url)

**Comment by pukkandan** [comment]
Ping me when this is done

**Comment by HobbyistDev** [comment]
@pukkandan, i think this fine. Can you review this ?

File: yt_dlp/extractor/rtllu.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
+from .common import InfoExtractor
+
+
+class RTLLuBaseIE(InfoExtractor):
+    _MEDIA_REGEX = {
+        'video': r'<rtl-player\s*poster\s*=\s*\"(?:[\"\w\.://-]+)\"\s*(title\s*=\s*(?:[\"\w\.-]+))?\s*(type\s*=\s*\"(?:\w+)\")?\s*(channelname\s*=\s*\"(?:\w+)\")?\s*hls\s*=\s*\"(?P<media_url>[\w\.\:/-]+)\"',
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/rtllu.py]: regex simpliified in https://github.com/yt-dlp/yt-dlp/pull/4222/commits/57fbc5da1c2e84609e087aa5012365a4e06b8988

**Pull request merged**